### PR TITLE
Fix EXTERNAL_PROPRIETARY implementation of commands.

### PR DESCRIPTION
**Pull request by vlantonov**
- Fixed implementation of hmi::GetUrls and mobile::OnSystemRequestNotification in case of defined EXTERNAL_PROPRIETARY.
- Improved memory handling of HandleOutgoingMessageProtocolV2.    
- Added minor fixes in logic and log messages.
- Fixed PolicyHandler and SubscribeWayPointsRequest tests.

Related Issue: [APPLINK-30609](https://adc.luxoft.com/jira/browse/APPLINK-30609)

### CODE REVIEW

**Comment by vlantonov** [comment]
Please review: @AByzhynar , @AGaliuzov , @OHerasym , @Kozoriz , @wolfylambova22

File: src/components/application_manager/src/mobile_message_handler.cc [hunk 1]
```diff
@@ -232,7 +232,7 @@ MobileMessageHandler::HandleOutgoingMessageProtocolV2(
 
   const size_t dataForSendingSize =
       protocol_handler::PROTOCOL_HEADER_V2_SIZE + jsonSize + binarySize;
-  uint8_t* dataForSending = new uint8_t[dataForSendingSize];
+  std::vector<uint8_t> dataForSending(dataForSendingSize);
```
Comment (AByzhynar) [inline src/components/application_manager/src/mobile_message_handler.cc]: @vlantonov We have a typedef for this type of data, please use it. I assume it called BinaryData

**Comment by AByzhynar** [comment]
@dev-gh optional: Please also review

File: src/components/application_manager/test/commands/mobile/subscribe_way_points_request_test.cc [hunk 2]
```diff
@@ -90,27 +93,38 @@ TEST_F(SubscribeWayPointsRequestTest, Run_SUCCESS) {
 }
 
 TEST_F(SubscribeWayPointsRequestTest, OnEvent_SUCCESS) {
+  const hmi_apis::Common_Result::eType result_code =
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/subscribe_way_points_request_test.cc]: @vlantonov 
Move it to the place before its first usage

**Comment by vlantonov** [comment]
@dev-gh Actually these changes do no change the lo logic as they are minor refactoring. They were introduced during the investigation of the process flow and took little time. The established practice is to do minor fixes when possible to be done quickly.

File: src/components/application_manager/src/mobile_message_handler.cc (hunk 1 above)
Comment (vlantonov) [inline src/components/application_manager/src/mobile_message_handler.cc]: @AByzhynar Fixed in d97f9aaaa3701bf84c1eee7256cec3e4610bc122

File: src/components/application_manager/test/commands/mobile/subscribe_way_points_request_test.cc (hunk 2 above)
Comment (vlantonov) [inline src/components/application_manager/test/commands/mobile/subscribe_way_points_request_test.cc]: @AByzhynar Fixed in d97f9aaaa3701bf84c1eee7256cec3e4610bc122

**Comment by ovvasyliev** [comment]
Rebuild required

File: src/components/application_manager/src/mobile_message_handler.cc [hunk 3]
```diff
@@ -206,13 +206,11 @@ MobileMessageHandler::HandleOutgoingMessageProtocolV1(
     return NULL;
   }
 
-  uint8_t* rawMessage = new uint8_t[messageString.length() + 1];
-  memcpy(rawMessage, messageString.c_str(), messageString.length() + 1);
+  BinaryData rawMessage(messageString.length() + 1);
```
Comment (AByzhynar) [inline src/components/application_manager/src/mobile_message_handler.cc]: @vlantonov I would just rename variable acc. Google Coding Style

File: src/components/application_manager/src/mobile_message_handler.cc [hunk 4]
```diff
@@ -232,7 +230,7 @@ MobileMessageHandler::HandleOutgoingMessageProtocolV2(
 
   const size_t dataForSendingSize =
       protocol_handler::PROTOCOL_HEADER_V2_SIZE + jsonSize + binarySize;
-  uint8_t* dataForSending = new uint8_t[dataForSendingSize];
+  BinaryData dataForSending(dataForSendingSize);
```
Comment (AByzhynar) [inline src/components/application_manager/src/mobile_message_handler.cc]: @vlantonov I would just rename variables acc. Google Coding Style

**Comment by ovvasyliev** [comment]
Rebuild required

**Comment by antongovorov** [comment]
Rebuild required

**Comment by vlantonov** [comment]
Rebuild required

**Comment by vlantonov** [comment]
Rebuild required

**Comment by vlantonov** [comment]
Rebuild required

**Pull request merged**